chalk-ir/src/fold.rs

```diff
@@ -474,6 +474,9 @@ macro_rules! copy_fold {
 copy_fold!(Identifier);
 copy_fold!(UniverseIndex);
 copy_fold!(ItemId);
+copy_fold!(StructId);
```

Comment:
note that rust-analyzer distinguishes further between structs and enums, and the various language-defined types (primitives, function pointers etc.). the distinction doesn't matter to Chalk, but it matters a lot in rust-analyzer :) we'll have to think about how to handle this.

Reply:
Interesting.